src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -71,7 +71,7 @@ static int smart_amp_init(struct processing_module *mod)
 
 	/* Copy the pin formats */
 	bs = sizeof(sad->ipc4_cfg.input_pins) + sizeof(sad->ipc4_cfg.output_pin);
-	memcpy_s(sad->ipc4_cfg.input_pins, bs,
+	memcpy_s(&sad->ipc4_cfg.input_pins[0], bs,
```

Comment:
Leaving the specific analyzer aside, can you explain in the commit message what difference this change makes with respect to buffer overflows?

Reply:
Just realized this is copying two contiguous fields at once. No surprise the static analyzer is not happy!

It's also a bit strange to use `memcpy_s()` whose only purpose is to force the developer to be extra careful and then use it to "overflow" the copy from one field to the other.

I would either do two copies or use `memcpy`, not `memcpy_s`.

Reply:
I agree there should be separate copies for the input and out pin formats.

Reply:
Done, thanks.